tidy3d/plugins/mode/solver.py

```diff
@@ -22,6 +22,8 @@
 TOL_TENSORIAL = 1e-6
 # shift target neff by this value, both rel and abs, whichever results in larger shift
 TARGET_SHIFT = 10 * fp_eps
+# Preconditioner: "Jacobi" or "Material" based
+PRECONDITIONER = "Material"
```

Comment:
Now I also include the previous preconditioning method (Jacobi) along with the new one. For non-magnetic material (mu=1 everywhere), the two are actually very similar. The new one performs better when both epsilon and mu can be huge.